src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4670,12 +4726,20 @@ private static bool CanReuseParameter(CSharp.Syntax.ParameterSyntax parameter)
                 }
             }
 
+            // cannot reuse parameters without identifiers (found in extension declarations) as they are parsed with different rules.
+            // The reverse is fine though.  Normal parameters (from non extensions) can be re-used into an extension declaration
+            // as all normal parameters are legal extension parameters.
+            if (parameter.Identifier.Kind() == SyntaxKind.None)
+            {
+                return false;
```

Comment:
>return false [](http://example.com/codeflow?start=16&length=12)

It looks like in every place where the enclosing method is called, we know whether missing identifier is allowed. We can pass this information and take advantage of that here.